src/izgbs.py

```diff
@@ -151,34 +167,46 @@ def izgbs(
         max_wait_time_std = np.std(max_wait_times)
 
         # populate results
-        feasible_df.loc[feasible_df.Machines == num_machines, 'BatchAvg'] = avg_wait_time_avg
-        feasible_df.loc[feasible_df.Machines == num_machines, 'BatchMaxAvg'] = max_wait_time_avg
+        if num_machines in feasible_dict:
+            feasible_dict[num_machines]['BatchAvg'] = avg_wait_time_avg
+            feasible_dict[num_machines]['BatchMaxAvg'] = max_wait_time_avg
```

Comment:
Is this 'in' check needed? If it's not then delete the if statement (keep the body)